python/pyspark/worker.py

```diff
@@ -253,7 +253,9 @@ def read_udfs(pickleSer, infile, eval_type):
             "spark.sql.legacy.execution.pandas.groupedMap.assignColumnsByName", "true")\
             .lower() == "true"
 
-        ser = ArrowStreamPandasUDFSerializer(timezone, safecheck, assign_cols_by_name)
+        df_for_struct = eval_type == PythonEvalType.SQL_SCALAR_PANDAS_UDF
```

Comment:
It seems hard to tell why when `eval_type` is `PythonEvalType.SQL_SCALAR_PANDAS_UDF`, then `df_for_struct` should be true. Maybe a well explained comment here is better.

Reply:
Sure, will add a comment.